src/components/Sidebar/Tabs/CustomizationTab/MarkSubTab.tsx

```diff
@@ -0,0 +1,143 @@
+/** @jsx jsx */
+import { jsx, css } from '@emotion/react';
+import { useState } from 'react';
+
+import {
+  vegaCategoricalChartList,
+  vegaChartList,
+  vegaTemporalChartList,
+  preprocessEncoding,
+  convertToCategoricalChartsEncoding,
+} from '../../../../modules/VegaEncodings';
+import SearchBar from '../../../ui-widgets/SearchBar';
+import { CustomizeSubTapProps } from './CustomizationTab';
+import { GraphSpec, useModelState } from '../../../../hooks/bifrost-model';
+import { PlainObject, VegaLite, VisualizationSpec } from 'react-vega';
+import produce from 'immer';
+import theme from '../../../../theme';
+
+const markOptionsListCss = css`
+  display: flex;
+  flex-wrap: wrap;
+  overflow: scroll;
+  list-style: none;
+  padding: 0px;
+`;
+
+const markOptionCss = css`
+  display: flex;
+  flex-direction: column;
+
+  &.selected {
+    border: 5px solid ${theme.color.primary.light};
+
+    span {
+      font-weight: bold;
+    }
+  }
+`;
+
+export default function MarkSubTab(props: CustomizeSubTapProps) {
+  const [searchValue, setSearchValue] = useState<string>('');
+  const [results, setResults] = useState(
+    vegaChartList.map((choice, index) => ({ choice, index }))
+  );
+  const data = useModelState<PlainObject>('graph_data', (data) => ({
+    data,
+  }))[0];
+
+  const [selectedMark, setSelectedMark] = useState<
+    string | Record<string, any>
+  >(props.spec.mark);
+
+  function handleOnClick(spec: GraphSpec) {
+    const newSpec = produce(spec, (draftSpec: GraphSpec) => {
+      draftSpec['width'] = props.spec.width;
+      draftSpec['height'] = props.spec.height;
+    });
+
+    if (typeof newSpec.mark === 'object') {
+      setSelectedMark(newSpec.mark.type);
+    } else {
+      setSelectedMark(newSpec.mark);
+    }
+    props.setSpec(newSpec);
+  }
+
+  return (
+    <div className="mark-options-wrapper" style={{ width: '100%' }}>
+      <SearchBar
+        choices={vegaChartList}
+        onResultsChange={setResults}
+        onChange={setSearchValue}
+        value={searchValue}
+      />
+      <ul className="mark-options" css={markOptionsListCss}>
+        {results
+          .filter(({ choice: kind }) => {
+            const spec = produce(props.spec, (draftSpec: GraphSpec) => {
+              preprocessEncoding(draftSpec);
+            });
+
+            if (vegaCategoricalChartList.includes(kind)) {
+              const x = spec['encoding']['x'];
+              const y = spec['encoding']['y'];
+              if (
+                (x['type'] === 'quantitative' && y['type'] === 'nominal') ||
+                (y['type'] === 'quantitative' && x['type'] === 'nominal')
+              ) {
+                return true;
+              } else {
+                return false;
+              }
```

Comment:
replace this with `return (x['type'] === 'quantitative' && y['type'] === 'nominal') ||
                (y['type'] === 'quantitative' && x['type'] === 'nominal')`